newlib/libc/stdlib/strtol.c

```diff
@@ -123,6 +123,7 @@ No supporting OS subroutines are required.
 #include <errno.h>
 #include <stdlib.h>
 #include <reent.h>
+#include "../machine/xtensa/pgmspace.h"
```

Comment:
Why is it necessary to include pgmspace in this file?

Reply:
When building newlib -Os the ctype functions evaluate to the macros, and those require pgm_read_byte.  This function uses isupper() and isdigit().

Reply:
Does that mean that every user of <ctype.h> now has to include this machine-specific header file? How about including the necessary header file from ctype.h?

Reply:
Technically, yes, but practically for the Arduino SDK at least, no, because the one in the Arduino SDK is always included beforehand....argh....well, in the main .ino anyway.  Extra files do not get the same #include pre-treatment by the IDE/SDK if my quick and dirty test is to be believed.  So, one more vote for includeing the pgm_read_byte() stuff somewhere in newlib proper...

Will try to get something consolidated with this in a few days.